### PR TITLE
Feature/dom updates test

### PR DESCRIPTION
**Pull request by jeremiahblackol**
# PR Description
Please include a summary of the change:
This allows us to use spies to test DOM manipulation.

Commit Emojis:

:tada: Initial commit
:construction: HTML modifications/updates.
:art: CSS modifications/updates.
:crystal_ball: JS modifications/updates.
:lipstick: hotfix any code.
:bar_chart: new tests
:camera_flash: adding screenshots
:bento: Adding new assets or images
:beetle: If it is a bug or issue fix, describe the issue that was fixed.

If it is a new feature, describe the feature and the iteration of the project spec it addresses.
use of spies 

Note any questions or struggles as well.

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [ ] My code follows the Turing style guidelines
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: test/domUpdates-test.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
+/* eslint-disable max-len */
+const chai = require('chai');
```
Comment (Billwilke42) [inline test/domUpdates-test.js:2]: Nice work on this!

File: src/domUpdates.js [hunk 2]
```diff
@@ -1,55 +1,39 @@
...
-let favButton = document.querySelector('.view-favorites');
-let numberDropDown = document.querySelector('.number-dropdown');
 
 class DomUpdates {
   constructor() {
 
   }
 
-  viewFavorites(user, cookbook) {
-    if (cardArea.classList.contains('all')) {
```
Comment (Melizzo) [inline src/domUpdates.js:27]: Great solution - having a param for the area.

File: src/scripts.js [hunk 3]
```diff
@@ -68,56 +69,51 @@ favButton.addEventListener('click', viewFavorites);
...
+    let numberInput = document.getElementById('number-input')
+    postNewIngredientsData(ingredientsDropDown.value, numberInput.value)
+  } else if (event.target.classList.contains('card-picture')) {
+    domUpdates.displayDirections(event, cookbook, ingredientsData, pantry, cardArea);
   } 
 }
 
 function  viewRecipesToCook() {
-  domUpdates.viewRecipesToCook(user) 
+  domUpdates.viewRecipesToCook(user, cardArea, addedRecipeButton, cookbook)
```
Comment (Billwilke42) [inline src/scripts.js:108]: It's cool how were passing in the buttons now!

File: test/domUpdates-test.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    chai.spy.on(domUpdates, "filterRecipesByTag", () => {});
+
+    cookbook = new Cookbook(recipeData)
+    recipe = new Recipe()
+    user = new User(users[0].id, users[0].name, users[0].pantry)
+   
+  })
+
+
+  it('should be a function', function() {
```
Comment (Melizzo) [inline test/domUpdates-test.js:42]: absolutely great work to get the domUpdates going Jeremiah.

**Pull request merged**